File: python/matrix_multiply_2d/function.py
def matrix_multiply_2d(matrix1: list[list[int]], matrix2: list[list[int]]) -> list[list[int]]:
    if len(matrix1[0]) != len(matrix2):
        raise ValueError("The number of columns in matrix1 must be equal to the number of rows in matrix2.")

    
    result = [[0 for _ in range(len(matrix2[0]))] for _ in range(len(matrix1))]

    for i in range(len(matrix1)):
        for j in range(len(matrix2[0])):
            for k in range(len(matrix2)):
                result[i][j] += matrix1[i][k] * matrix2[k][j]

    return result

File: javascript/matrixMultiply2d/function.js
function matrixMultiply2d(matrix1, matrix2) {
    if (matrix1[0].length !== matrix2.length) {
        throw new Error("The number of columns in matrix1 must be equal to the number of rows in matrix2.");
    }

    const result = Array(matrix1.length).fill().map(() => Array(matrix2[0].length).fill(0));
    for (let i = 0; i < matrix1.length; i++) {
        for (let j = 0; j < matrix2[0].length; j++) {
            for (let k = 0; k < matrix2.length; k++) {
                result[i][j] += matrix1[i][k] * matrix2[k][j];
            }
        }
    }

    return result;
}
